source/s3_auto_ranged_get.c

```diff
@@ -691,6 +715,15 @@ static void s_s3_auto_ranged_get_request_finished(
                 auto_ranged_get->synced_data.head_object_completed = true;
                 AWS_LOGF_DEBUG(AWS_LS_S3_META_REQUEST, "id=%p Head object completed.", (void *)meta_request);
                 break;
+            case AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_PART_NUMBER:
+                auto_ranged_get->synced_data.get_first_part_completed = true;
+                AWS_LOGF_DEBUG(AWS_LS_S3_META_REQUEST, "id=%p Get Part Number completed.", (void *)meta_request);
+                if (error_code == AWS_ERROR_S3_GET_PART_SIZE_MISMATCH && found_object_size) {
+                    /* Try to fetch the first part again as a ranged get */
+                    break;
+                }
+                ++auto_ranged_get->synced_data.num_parts_requested;
```

Comment:
I made another comment elsewhere about "why don't we increment num_parts_requested when we create the GET_FIRST_PART?"

Now I see. But I think this would be less confusing if we DID increment num_parts_requested there in the normal way, and then over here if we realize that we need to re-request the part we do the weird thing like:
```
/* Try to fetch the first part again as a ranged get */
auto_ranged_get->synced_data.num_parts_requested = 0;
break;

Reply:
Thanks, fixed.